zerver/lib/message.py

```diff
@@ -835,7 +847,7 @@ def has_message_access(
 
 
 def bulk_access_messages(
-    user_profile: UserProfile, messages: Sequence[Message], *, stream: Optional[Stream] = None
+    user_profile: UserProfile, messages: Iterable[Message], *, stream: Optional[Stream] = None
```

Comment:
Prefer `messages: Collection[Message]` here because we iterate it twice. A call like `bulk_access_messages(user, (message for message in messages))` should be rejected, because although it won’t throw a type error, the generator will be unexpectedly empty for the second iteration.